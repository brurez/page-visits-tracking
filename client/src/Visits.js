import React, { Component } from 'react';
import axios from 'axios';
import moment from 'moment';

import LoadingMsg from './LoadingMsg';

class Visits extends Component {
  state = { data: [], isFetching: true };

  componentDidMount() {
    axios
      .get('/api/visits?sort=-visitedAt&limit=10')
      .then(res => {
        const { data } = res.data;
        this.setState({ data, isFetching: false });
      })
      .catch(err => console.log(err));
  }

  render() {
    const { data, isFetching } = this.state;
    if (!isFetching && data.length) {
      const visits = this.state.data.map(visit => {
        //debugger;
        return (
          <tr key={visit._id}>
            <td>{moment(visit.visitedAt).format('LLL')}</td>
            <td>{visit.url}</td>
            <td>{(visit._contact && visit._contact.email) || 'Anonimo'}</td>
          </tr>
        );
      });

      return (
        <table className="ui compact table">
          <thead>
            <tr>
              <th>Data da Visita</th>
              <th>Endereço da Página</th>
              <th>Contato</th>
            </tr>
          </thead>
          <tbody>{visits}</tbody>
        </table>
      );
    } else if (isFetching) {
      return <LoadingMsg />;
    } else {
      return (
        <div className="ui icon message">
          <i className="frown icon" />
          <div className="content">
            <div className="header">Nenhuma visita até agora</div>
            <p>
              Tenha um pouco de paciência, ainda vão querer visitar seu site.
            </p>
          </div>
        </div>
      );
    }
  }
}

export default Visits;
