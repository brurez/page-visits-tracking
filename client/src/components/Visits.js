import React, { Component } from 'react';
import axios from 'axios';
import moment from 'moment';

import LoadingMsg from './LoadingMsg';

class Visits extends Component {
  state = { data: [], isFetching: true, firstTime: true };

  componentDidMount() {
    this.fetchApi();
    this.interval = setInterval(() => {
      this.fetchApi();
    }, 6000);
  }

  fetchApi() {
    this.setState({ isFetching: true });
    axios
      .get('/api/visits?sort=-visitedAt&limit=10')
      .then(res => {
        const {  data } = res.data;
        this.setState({ data, isFetching: false, firstTime: false });
      })
      .catch(err => {
        console.log(err);
        this.fetchApi();
      });
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    const { data, isFetching, firstTime } = this.state;
    if (!firstTime && data.length) {
      const visits = this.state.data.map(visit => {
        //debugger;
        return (
          <tr key={visit._id}>
            <td>{moment(visit.visitedAt).format('LLL')}</td>
            <td>{visit.url}</td>
            <td>{(visit._contact && visit._contact.email) || 'Anonimo'}</td>
            <td className="collapsing" />
          </tr>
        );
      });

      const iconColor = isFetching ? 'inherit' : 'rgb(255, 255, 255)';

      return (
        <table className="ui striped compact table">
          <thead>
            <tr>
              <th>Data da Visita</th>
              <th>Endereço da Página</th>
              <th>Contato</th>
              <th>
                <i className="spinner loading transparent icon" style={{color: iconColor}}/>
              </th>
            </tr>
          </thead>
          <tbody>{visits}</tbody>
        </table>
      );
    } else if ( firstTime ) {
      return <LoadingMsg />;
    } else {
      return (
        <div className="ui icon message">
          <i className="frown icon" />
          <div className="content">
            <div className="header">Nenhuma visita até agora</div>
            <p>
              Tenha um pouco de paciência, ainda vão querer visitar seu site.
            </p>
          </div>
        </div>
      );
    }
  }
}

export default Visits;
