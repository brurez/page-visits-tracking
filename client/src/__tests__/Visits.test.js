import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import Visits from '../components/Visits';

jest.mock('axios');

configure({ adapter: new Adapter() });

describe('Contact Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Visits />);
  });

  test('if exist', () => {
    expect(wrapper).toBeTruthy();
  });

  test('if loading message is been displayed', () => {
    expect(wrapper.html()).toContain('Espere um segundo');
  });

  //update

  test('if display data from api', () => {
    wrapper.update();
    expect(wrapper.html()).toContain('rd.toplayalong.com/sobre.html');
    expect(wrapper.html()).toContain('test.com');
  });

  test('if match the snapshot', () => {
    wrapper.update();
    wrapper.update();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

