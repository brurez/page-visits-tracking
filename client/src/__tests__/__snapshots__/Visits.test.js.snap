// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component if match the snapshot 1`] = `
<table
  className="ui striped compact table"
>
  <thead>
    <tr>
      <th>
        Data da Visita
      </th>
      <th>
        Endereço da Página
      </th>
      <th>
        Contato
      </th>
      <th>
        <i
          className="spinner loading transparent icon"
          style={
            Object {
              "color": "rgb(255, 255, 255)",
            }
          }
        />
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="5a3fcc190e5e9300145a116e"
    >
      <td>
        December 24, 2017 1:47 PM
      </td>
      <td>
        rd.toplayalong.com/sobre.html
      </td>
      <td>
        Anonimo
      </td>
      <td
        className="collapsing"
      />
    </tr>
    <tr
      key="5a3fcc170e5e9300145a116d"
    >
      <td>
        December 24, 2017 1:47 PM
      </td>
      <td>
        test.com
      </td>
      <td>
        Anonimo
      </td>
      <td
        className="collapsing"
      />
    </tr>
  </tbody>
</table>
`;
